std/node/_fs/_fs_mkdtemp.ts

```diff
@@ -0,0 +1,92 @@
+// Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
+import { existsSync } from "./_fs_exists.ts";
+import {
+  ERR_INVALID_CALLBACK,
+  ERR_INVALID_OPT_VALUE_ENCODING,
+} from "../_errors.ts";
+
+export type mkdtempCallback = (
+  err: Error | undefined,
+  directory?: string,
+) => void;
+
+// https://nodejs.org/dist/latest-v15.x/docs/api/fs.html#fs_fs_mkdtemp_prefix_options_callback
+export function mkdtemp(prefix: string, callback: mkdtempCallback): void;
+export function mkdtemp(
+  prefix: string,
+  options: { encoding: string } | string,
+  callback: mkdtempCallback,
+): void;
+export function mkdtemp(
+  prefix: string,
+  optionsOrCallback: { encoding: string } | string | mkdtempCallback,
+  maybeCallback?: mkdtempCallback,
+): void {
+  const callback: mkdtempCallback | undefined =
+    optionsOrCallback instanceof Function ? optionsOrCallback : maybeCallback;
+  if (!callback) throw new ERR_INVALID_CALLBACK(callback);
+
+  const encoding: string | undefined = parseEncoding(optionsOrCallback);
+  const path = tempDirPath(prefix);
+
+  Deno.mkdir(path, { recursive: false, mode: 0o700 })
+    .then(() => callback(undefined, decode(path, encoding)))
```

Comment:
This patterns is problematic, if `callback` throws you reach the catch and the callback is called again with an error - doing `.then(() => callback(...), callback)` is probably safer or using `util.callbackify` if Deno already has it.

Reply:
It seems like the problematic pattern is used in many other compatibility implementations... __fs_appendFile, _fs_chmod, _fs_chown, _fs_copy, _fs_dir, _fs_exists, _fs_exists, _fs_lstat, _fs_mkdir_ ... to only name a few.

I don't understand what is the safer solution proposal ` .then(() => callback(...), callback)` ?

Reply:
Yes, and then it is also wrong and very dangerous in all those places 😅

Reply:
Has any ticket been opened about that improvement?

Reply:
I don't know I'm not a maintainer, just a contributor that responded to the "please review this" ping on discord since I'm a node maintainer - maybe Bert would know :]

Reply:
I'll fill in a ticket about this.  Thanks for the insight @benjamingr.